src/charactersheet/models/dm/encounter_sections/encounter_armor.js

```diff
@@ -1,38 +1,143 @@
+            return '';
+        }
+    });
+
+    armorDescriptionHTML = ko.pureComputed(() => {
+        if (this.description()){
+            return this.description().replace(/\n/g, '<br />');
+        } else {
+            return '<div class="h3"><small>Add a description via the edit tab.</small></div>';
+        }
+    });
+
+    armorSummaryLabel = ko.pureComputed(() => {
+        if (this.magicalModifier() != 0) {
+            if (this.acLabel()) {
+                return this.magicalModifierLabel() + ', ' + this.acLabel();
+            } else {
+                return this.magicalModifierLabel();
+            }
+        } else {
+            return this.acLabel();
+        }
+    });
+
+    magicalModifierLabel = ko.pureComputed(() => {
+        var magicalModifier = this.magicalModifier();
+        if (magicalModifier != 0) {
+            return magicalModifier >= 0 ? ('+ ' + magicalModifier) : '- ' +
+            Math.abs(magicalModifier);
+        } else {
+            return '';
+        }
     });
 
-    self.propertyLabel = ko.pureComputed(function() {
-        return self.armorClass() ? self.acLabel() : '';
+    armorWeightLabel = ko.pureComputed(() => {
+        return this.weight() !== '' && this.weight() >= 0 ? this.weight() + ' lbs.' : '0 lbs.';
     });
 
-    self.descriptionLabel = ko.pureComputed(function() {
-        return self.shortDescription();
+    nameLabel = ko.pureComputed(() => {
+        return this.name();
     });
 
-    // UI Methods
-    self.shortDescription = ko.pureComputed(function() {
-        return Utility.string.truncateStringAtLength(self.armorDescription(), self.SHORT_DESCRIPTION_MAX_LENGTH);
+    propertyLabel = ko.pureComputed(() => {
+        return this.armorClass() ? this.acLabel() : '';
     });
 
-    return self;
-}
-EncounterArmor.__name = 'EncounterArmor';
+    descriptionLabel = ko.pureComputed(() => {
+        return this.shortDescription();
+    });
+
+    shortDescription = ko.pureComputed(() => {
+        return Utility.string.truncateStringAtLength(this.description(), this.SHORT_DESCRIPTION_MAX_LENGTH);
+    });
+
+    clean = (keys, params) => {
+        let treasure = pick(params, EncounterArmor.mapping.include);
+        treasure.value = pick(params, EncounterArmor.armorFields);
+        return treasure;
+    };
+
+    buildModelFromValues = (values) => {
+        let keys = Object.keys(values);
+        keys.forEach((key) => {
+            if (key === 'type') {
+                this['armorType'] = values[key];
+            } else {
+                this[key] = values[key];
+            }
+        });
+    };
+
+    getValues = () => {
+        let values = {};
+        EncounterArmor.armorFields.forEach((field) => {
+            if (field === 'armorType') {
+                values['type'] = this[field];
+            } else {
+                values[field] = this[field];
+            }
+        });
+
+        return values;
+    };
+
+    customExportValues = () => {
```

Comment:
The names for these methods `custom.*` is a bit ambiguous.

How about something like:
`toJSON or fromJSON`

Either way, can you document what the purpose of these methods are? This can be copy-pasted to all of them, but a quick blurb about why you don't use the built-in ones would eliminate future confusion.

i.e.
> Serialize the current item to a plain JSON format. We use these in-leiu of the normal `import/exportValues` because those return a format unsuitable for reimporting (since it caused data corruption).